experimental/InjectiveResolutions/src/LocalCohomology.jl

```diff
@@ -0,0 +1,642 @@
+
+!!! note
+    The monoid algebra $k[Q]$ must be normal.  
+"""
+function local_cohomology_all(I_M::MonoidAlgebraIdeal, I::MonoidAlgebraIdeal, i::Integer)
+  @req I_M.algebra == I.algebra "ideals must be over same monoid algebra"
+  return local_cohomology_all(quotient_ring_as_module(M),I,i)
+end
+
+@doc raw"""
+    local_cohomology_all(M::SubquoModule{T}, I::MonoidAlgebraIdeal, i::Integer) where {T<:MonoidAlgebraElem}
+
+For $1 \leq j \leq i $ compute sector partitions of the local cohomology modules $H^i_I(M))$. This function only computes one injective resolution of $M$ up to cohomological degree $i+1$. 
+
+!!! note
+    The monoid algebra $k[Q]$ must be normal.  
+"""
+function local_cohomology_all(M::SubquoModule{T}, I::MonoidAlgebraIdeal, i::Integer) where {T<:MonoidAlgebraElem}
+  kQ = base_ring(M)
+  @req kQ == I.monoid_algebra "module and ideal must be over same monoid algebra"
+  @req is_normal(kQ) "monoid algebra must be normal"
+
+  #compute injective resolution
+  inj_res = injective_resolution(M, i+1)
+
+  #compute a sector partition of H^j_I(M) for 1 \leq j \leq i
+  H = Vector{SectorPartitionLC}()
+  for j in 1:i
+    _phi = get_scalar_matrix(kQ, inj_res.cochain_maps[j])
+    _psi = get_scalar_matrix(kQ, inj_res.cochain_maps[j + 1])
+    Jj = inj_res.inj_mods[j]
+    Jj_1 = inj_res.inj_mods[j + 1]
+    Jj_2 = inj_res.inj_mods[j + 2]
+
+    J, phi, psi, (j, k) = apply_gamma!(Jj, Jj_1, Jj_2, _phi, _psi, I)
+    Hj = sector_partition(kQ, phi, psi, j, k, J...)
+    push!(
+      H,
+      SectorPartitionLC(M, j, I.ideal, Hj, maps_needed(kQ, Hj)),
+    )
+  end
+  return H
+end
+
+# given a matrix with entries in a MPolyDecRing or a MPolyQuoRing, return the matrix with the corresponding scalar coefficients
+@doc raw"""
+  get_scalar_matrix(A::MonoidAlgebra, M)
+
+Given a monomial matrix, return the corresponding coefficient matrix. 
+"""
+function get_scalar_matrix(A::MonoidAlgebra, M)
+  k = coefficient_ring(A)
+  return map(x -> evaluate(x,ones(elem_type(k),ngens(A))), M)
+end
+
+# given a monoid algebra k[Q] compute linear functions \tau_1,...,\tau_n that define Q. 
+# there is one defining linear functional for every facet of Q and Q is the intersection of the halfspaces {\tau_i \leq 0} 
+
+@doc raw"""
+  compute_taus(kQ::MonoidAlgebra, J::IndecInj...)
+
+This function computes a vector $\tau^j \in (\mathbb{Z}\cup \infty)^n$ for every indecomposable injective $J^j$.
+See [HM05](@cite) (Section 6).
+"""
+function compute_taus(kQ::MonoidAlgebra, J::IndecInj...)
+  # get linear functionals tau_i for every hyperplane bounding Q, i.e. for every facet
+  A, _ = halfspace_matrix_pair(facets(kQ.cone))
+  H = []
+  for i in 1:length(kQ.hyperplanes)
+    push!(H, [kQ.hyperplanes[i].hyperplane, A[i, :]])
+  end
```

Comment:
It may not be too relevant in this function (I can't judge it), but be warned that whenever you use `[]` you are creating a `Vector{Any}` which tends to prevent type stability later in the function. Here at least it is easy enough to fix:
```suggestion
  H = [[kQ.hyperplanes[i].hyperplane, A[i, :]] for i in 1:length(kQ.hyperplanes)]
```

Reply:
(Thought it might be better to just call `get_halfspace_eq` here?)